### PR TITLE
Adding cuda-gdb recipe

### PR DESCRIPTION
**Pull request by adibbley**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/cuda-gdb/build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+# Install to conda style directories
+[[ -d lib64 ]] && mv lib64 lib
+
+for i in `ls`; do
+    [[ $i == "build_env_setup.sh" ]] && continue
+    [[ $i == "conda_build.sh" ]] && continue
+    [[ $i == "metadata_conda_debug.yaml" ]] && continue
```
Comment (bdice) [inline recipes/cuda-gdb/build.sh:9]: Are these still relevant?

Comment (adibbley) [reply recipes/cuda-gdb/build.sh:9]: Yes, due to the lazy copying we do in these scripts. Tested without and they are picked up:
```
‘build_env_setup.sh’ -> ‘$PREFIX/build_env_setup.sh’
‘conda_build.sh’ -> ‘$PREFIX/conda_build.sh’
‘metadata_conda_debug.yaml’ -> ‘$PREFIX/metadata_conda_debug.yaml’
```

File: recipes/cuda-gdb/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        - test -f $PREFIX/extras/cuda-gdb-{{ version }}.src.tar.gz
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: CUDA-GDB is the NVIDIA tool for debugging CUDA applications
+      description: |
+        CUDA-GDB is the NVIDIA tool for debugging CUDA applications running on Linux
+        and Mac. CUDA-GDB is an extension to the x86-64 port of GDB, the GNU Project
```
[the remark this answers is not in the document]
Comment (adibbley) [reply recipes/cuda-gdb/meta.yaml]: Looks like it is no longer accurate. IIRC Devtools supported Mac longer than other components, so at one point it was accurate to say `cuda-gdb` supported Mac, while our installers did not.

**Review by bdice — APPROVED** [review]
Looks fine to me!

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**